src/primitives/transaction.cpp

```diff
@@ -16,7 +16,9 @@ JSDescription::JSDescription(ZCJoinSplit& params,
             const boost::array<libzcash::JSOutput, ZC_NUM_JS_OUTPUTS>& outputs,
             CAmount vpub_old,
             CAmount vpub_new,
-            bool computeProof) : vpub_old(vpub_old), vpub_new(vpub_new), anchor(anchor)
+            bool computeProof,
+            uint256 *esk // payment disclosure
```

Comment:
Does this need to be deleted in a destructor somehwere?

Reply:
No, esk is a stack variable and will be destroyed when it falls out of scope. JSDescription expects a pointer to esk, so that it can later dereference and set the value of the symmetric key.

Reply:
But does that mean that the thing esk is pointing to is cleared from the heap when esk is destroyed?  Doesn't that only happen with unique pointers?